src/Modal.js

```diff
@@ -161,7 +161,11 @@ const Modal = React.createClass({
     let show = !!props.show;
     let dialog = React.Children.only(this.props.children);
 
-    let setMountNode = ref => this.mountNode = (!ref || ref.getMountNode());
+    let setMountNode = ref => {
+      if (ref !== null && typeof ref === 'object') {
```

Comment:
Sorry @craigbilner what do you mean "move it into the class"? What problem does having a new function cause on each render?